src/main/java/io/cdap/cdap/events/process/pubsub/PubSubEventWriter.java

```diff
@@ -0,0 +1,201 @@
+        }
+    }
+
+    /**
+     * Method to use the service account path as a File
+     *
+     * @param saPath Service account where is storage
+     * @return Service account as a file
+     * @throws FileNotFoundException If the file does not exists, it will throws a FileNotFoundException
+     */
+    private InputStream getCredentials(String saPath) throws FileNotFoundException {
+        File credentialsFile = new File(saPath);
+        return new FileInputStream(credentialsFile);
+    }
+
+    /**
+     * Method to configure proxy to the Pub/Sub Builder
+     *
+     * @param builder   Pub/Sub builder where was already initialized with the topic name, and project
+     * @param proxyHost Proxy host where configure the use of the proxy
+     * @param proxyPort Proxy port where configure the use of the proxy
+     * @throws NumberFormatException
+     */
+    private void configureChannelProxy(Publisher.Builder builder, String proxyHost, String proxyPort)
+            throws NumberFormatException {
+        SocketAddress proxySocketAddress = new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort));
+        builder.setChannelProvider(
+                InstantiatingGrpcChannelProvider.newBuilder()
+                        .setChannelConfigurator(managedChannelBuilder -> managedChannelBuilder.proxyDetector(
+                                new ProxyDetector() {
+                                    @Nullable
+                                    @Override
+                                    public ProxiedSocketAddress proxyFor(SocketAddress socketAddress) {
+                                        if (socketAddress == null) {
+                                            return null;
+                                        }
+                                        return HttpConnectProxiedSocketAddress.newBuilder()
+                                                .setTargetAddress((InetSocketAddress) socketAddress)
+                                                .setProxyAddress(proxySocketAddress)
+                                                .build();
+                                    }
+                                }))
+                        .build()
+        );
+    }
+
+    /**
+     *
+     * @return Instance Pub/sub publisher
+     */
+    protected Publisher getPublisher() {
+        return this.publisher;
+    }
+
+    @Override
+    public void write(Collection events) {
+        if (publisher == null) {
+            logger.debug("Publisher is not already initialized");
+            return;
+        }
+        Iterator iterator = events.iterator();
+
+        while (iterator.hasNext()) {
+            String stringEvent = GSON.toJson(iterator.next());
+            ByteString data = ByteString.copyFromUtf8(stringEvent);
+            PubsubMessage pubsubMessage = PubsubMessage.newBuilder().setData(data).build();
+
+            ApiFuture<String> future = this.publisher.publish(pubsubMessage);
```

Comment:
Is there a batch publish api available that can take more than one message?

Reply:
yes, I found the method: https://cloud.google.com/pubsub/docs/samples/pubsub-publisher-batch-settings#pubsub_publisher_batch_settings-java